infra/modules/azure_naming_convention/variables.tf

```diff
@@ -40,3 +40,78 @@ variable "environment" {
 
   description = "Values which are used to generate resource names and location short names. They are all mandatory except for domain, which should not be used only in the case of a resource used by multiple domains."
 }
+
+variable "resource_type" {
```

Comment:
why do we need this? can't we set this value inside tf modules?

Reply:
the module needs to know somehow what kind of resource it is, what do you mean by tf modules?

Reply:
Probably I'm missing something. Ehen you define some resource you either

a) use some terraform provider resource ie. `azurerm_cosmosdb_account`. if you have to also specify `resource_type = "Cosmos DB"` doesn't this results in a duplication ?
b) use some dx module which can specify `resource_type` itself

Reply:
Ah, okay, I think I understand what you mean. Yes, that makes sense, but a module doesn’t always create just the specific resource it is intended for. 
For example, some modules also create private endpoints, subnets, and storage accounts, so, if we used only the module as a reference, we wouldn’t be able to assign names to all the other resources

Reply:
I intended to handle the two cases differently (if possible):

a) Attempt to derive the resource type directly from the Terraform resource, avoiding the need for the user to specify the resource type manually.

b) Apply (a) for DX modules, or alternatively, include the resource type within the DX module resources ourselves.

Reply:
I have an idea on how we could improve this functionality. We could pass the module name as an input and generate all the required values as an output. 
For example, if we pass the module name `azure_cosmos_account`, the output could have a structure like this:
```
{
  "cosmos": "...",
  "private_endpoint": "..."
}
```
This way, each module would define its own set of names to be used

Alternatively, we could simplify it further by always generating the output with all possible abbreviations, and then the relevant one can be referenced. 
For example: `module.naming_convention.name.cosmos`

Reply:
Yes, that’s correct. However, the issue I see is that each module creates multiple resources with different names. For example, if the module context suggests it is for an `app_service`, it would need to generate names for the `app_service`, `private_endpoints`, `service_plan`, and `subnet`. This introduces additional complexity. By using a map with all the values as I’ve done now, the module can simply reference the appropriate names where needed (e.g., for `azurerm_app_service`, the `name` will be `module.naming_convention.names.app_service`).